boto/auth.py

```diff
@@ -1038,13 +1060,22 @@ def _wrapper(self):
         if boto.config.get('s3', 'use-sigv4', False):
             return ['hmac-v4-s3']
 
-        if hasattr(self, 'host'):
-            # If you're making changes here, you should also check
-            # ``boto/iam/connection.py``, as several things there are also
-            # endpoint-related.
-            for test in SIGV4_DETECT:
-                if test in self.host:
-                    return ['hmac-v4-s3']
+        if not hasattr(self, 'host'):
+            return func(self)
 
-        return func(self)
+        # Use default for the global region and non-aws hosts
+        if 's3.amazonaws.com' in self.host or 'amazonaws.com' not in self.host:
```

Comment:
I generated that using the endpoint resolver to find regions that were compatible with sigv2. It's a good point that `external-1` isn't in there so it should be added.